$MirOS: ports/net/djbdns/patches/patch-src_dnsname_c,v 1.1.201.1 2005/05/14 23:21:03 tg Exp $
--- src/dnsname.c.orig	Sun Feb 11 21:11:23 2001
+++ src/dnsname.c	Tue Dec 13 21:44:26 2005
@@ -2,6 +2,9 @@
 #include "exit.h"
 #include "strerr.h"
 #include "ip4.h"
+#ifdef DJB_V6ONLY
+#include "ip6.h"
+#endif /* DJB_V6ONLY */
 #include "dns.h"
 
 #define FATAL "dnsname: fatal: "
@@ -9,6 +12,9 @@
 static char seed[128];
 
 char ip[4];
+#ifdef DJB_V6ONLY
+char ip6[16];
+#endif /* DJB_V6ONLY */
 static stralloc out;
 
 int main(int argc,char **argv)
@@ -18,10 +24,22 @@ int main(int argc,char **argv)
   if (*argv) ++argv;
 
   while (*argv) {
+#ifndef DJB_V6ONLY
     if (!ip4_scan(*argv,ip))
       strerr_die3x(111,FATAL,"unable to parse IP address ",*argv);
     if (dns_name4(&out,ip) == -1)
       strerr_die4sys(111,FATAL,"unable to find host name for ",*argv,": ");
+#else
+    if (ip6_scan(*argv,ip6)) {
+      if (dns_name6(&out,ip6) == -1)
+	strerr_die4sys(111,FATAL,"unable to find host name for ",*argv,": ");
+    } else {
+      if (!ip4_scan(*argv,ip))
+	strerr_die3x(111,FATAL,"unable to parse IP address ",*argv);
+      if (dns_name4(&out,ip) == -1)
+	strerr_die4sys(111,FATAL,"unable to find host name for ",*argv,": ");
+    }
+#endif /* DJB_V6ONLY */
 
     buffer_put(buffer_1,out.s,out.len);
     buffer_puts(buffer_1,"\n");
